### PR TITLE
Blog App | Integrate Specs🎯

### PR DESCRIPTION
**Pull request by Rysth**
# 🏅 Features Implemented

List of things that were implemented based on the requirements:

## 📝 General requirements

- ✅ No Linters errors.
- ✅ Used GitFlow correctly.
- ✅ Created a professional `README` file.

## 📝 Project requirements

- ✅ Solved n+1 problems in the route `/users/:user_id/posts`.
- ✅ Installed Capybara gem to write integration tests.
- ✅ Added tests for `index` and `show` from Users.
- ✅ Added tests for `index` and `show` from Posts.

### N+1 (BEFORE)
before

### N + 1 (AFTER)
after

[_John Palacios_](https://www.linkedin.com/in/john-palacios-rysthcraft/)
_RysthCraft_

### CODE REVIEW

File: spec/system/users/show_spec.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  end
+
+  it 'Should render user bio' do
+    expect(page).to have_content('Specer')
+  end
+
+  it 'Should render user first 3 posts' do
+    expect(page).to have_selector('#user-photo', count: 1)
+  end
+
```
Comment (Gambit142) [inline spec/system/users/show_spec.rb:31]: - [ ] Great work writing this test case, but you are not checking if the page renders **_the first three comments_**. Remember that in a previous milestone, you created a method called `recent_post`. This method renders the **_three most recent posts_** and it was used on the `Users#Show` page. Kindly rewrite the test to reflect this. 💻 😎

File: spec/system/posts/index_spec.rb [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    expect(page).to have_content('Specing')
+  end
+
+  it 'Should render post text' do
+    expect(page).to have_content('Testing specs')
+  end
+
+  it 'Should render first comments on a post' do
+    expect(page).to have_content('Welcome World!')
+  end
```
Comment (Gambit142) [inline spec/system/posts/index_spec.rb:36]: - [OPTIONAL] I understand that you are creating only one comment for this post and in a real-world situation, that comment will be the only one to show if it is the only comment. However, I would like to suggest that you create more comments for this post and ensure that the `recent_comments` method works the way it should on the `Posts#Index` page. 👍 😎

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @Rysth @SnakyMz 👋, 

Good job so far!


### Highlights
- Setup linters correctly ✔️ 
- Wrote integration tests for `Users` controllers and view files ✔️ 
- Wrote integration tests for `Posts` controllers and view files ✔️

You still need to work on some issues to go to the next project, but you are almost there!

## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend considering them as they can improve your code._

You can also consider the following:

N/A

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### Please tag me in your question so I can receive the notification. You can tag me by typing @Gambit142

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: app/controllers/posts_controller.rb [hunk 3]
```diff
@@ -1,6 +1,6 @@
 class PostsController < ApplicationController
   def index
-    @user = User.find(params[:user_id])
+    @user = User.find(params[:user_id]) # 1
     @posts = @user.posts
   end
```
[the remark this answers is not in the document]
Comment (Rysth) [reply app/controllers/posts_controller.rb]: Trying my best, I hope that's Ok.

File: spec/system/users/show_spec.rb (hunk 1 above)
Comment (Rysth) [reply spec/system/users/show_spec.rb:31]: Done! Thanks!

File: spec/system/posts/index_spec.rb (hunk 2 above)
Comment (Rysth) [reply spec/system/posts/index_spec.rb]: Done!

**Review by brainconnect93 — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi @Rysth & @SnakyMz, 👋 👋  

Your project is complete! Great job for following the project requirement🥇and implementing the changes requested by the previous reviewer. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 




## Good Points 👍

- Descriptive Pull Request title & summary. ✔️
- integration specs for views and fix n+1 problems ✔️
- Good use of Git flow ✔️

## Optional changes

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brainconnect93 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**